File: servico-musica-java/java.grpc/src/main/java/servico/java/grpc/controller/UsuarioController.java
package servico.java.grpc.controller;

import io.grpc.stub.StreamObserver;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.grpc.server.service.GrpcService;

import servico.java.grpc.entity.Usuario;
import servico.java.grpc.proto.UsuarioListResponse;
import servico.java.grpc.proto.UsuarioRequest;
import servico.java.grpc.proto.UsuarioResponse;
import servico.java.grpc.proto.UsuarioServiceGrpc;
import servico.java.grpc.repository.UsuarioRepository;

import java.util.List;

@GrpcService
public class UsuarioController extends UsuarioServiceGrpc.UsuarioServiceImplBase {

    @Autowired
    private UsuarioRepository usuarioRepository;

    @Override
    public void todosUsuarios(UsuarioRequest request, StreamObserver<UsuarioListResponse> responseObserver) {

        List<Usuario> usuarios = usuarioRepository.findAll();

        UsuarioListResponse.Builder response = UsuarioListResponse.newBuilder();

        for (Usuario usuario : usuarios) {
            response.addUsuarios(
                    UsuarioResponse.newBuilder()
                            .setId(usuario.getId())
                            .setNome(usuario.getNome())
                            .setIdade(usuario.getIdade())
                            .build()
            );
        }

        responseObserver.onNext(response.build());
        responseObserver.onCompleted();
    }
}

File: servico-musica-kotlin/kotlin.grpc/src/main/kotlin/servico/kotlin/grpc/controller/UsuarioController.kt
package servico.kotlin.grpc.controller

import io.grpc.stub.StreamObserver
import org.springframework.grpc.server.service.GrpcService
import servico.kotlin.grpc.entity.Usuario
import servico.kotlin.grpc.proto.UsuarioListResponse
import servico.kotlin.grpc.proto.UsuarioRequest
import servico.kotlin.grpc.proto.UsuarioResponse
import servico.kotlin.grpc.proto.UsuarioServiceGrpc
import servico.kotlin.grpc.repository.UsuarioRepository

@GrpcService
class UsuarioController(
    private val usuarioRepository: UsuarioRepository
): UsuarioServiceGrpc.UsuarioServiceImplBase() {

    override fun todosUsuarios(
        request: UsuarioRequest?,
        responseObserver: StreamObserver<UsuarioListResponse?>
    ) {
        val usuarios: MutableList<Usuario> = usuarioRepository.findAll()

        val response: UsuarioListResponse.Builder =
            UsuarioListResponse.newBuilder()

        for (usuario in usuarios) {
            response.addUsuarios(
                UsuarioResponse.newBuilder()
                    .setId(usuario.id)
                    .setNome(usuario.nome)
                    .setIdade(usuario.idade.toInt())
                    .build()
            )
        }

        responseObserver.onNext(response.build())
        responseObserver.onCompleted()
    }
}